flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -381,7 +381,8 @@ private boolean isPartialResourceConfigured(JobGraph jobGraph) {
         return persistAndRunFuture.handleAsync(
                 (acknowledge, throwable) -> {
                     if (throwable != null) {
-                        cleanUpJobData(jobGraph.getJobID(), true);
+                        final boolean jobGraphRemoved = cleanUpJobGraph(jobGraph.getJobID(), true);
+                        cleanUpRemainingJobData(jobGraph.getJobID(), true, jobGraphRemoved);
```

Comment:
Maybe we can introduce a helper method for this to give this operation a good name.

Reply:
👍